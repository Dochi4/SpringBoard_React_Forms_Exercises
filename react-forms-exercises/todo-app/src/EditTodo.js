import React, { useState } from "react";

const EditTodo = ({
  id,
  item,
  completed,
  toggleTaskCompleted,
  editTodo,
  removeTodo,
}) => {
  const [isEditing, setEditing] = useState(false);
  const [formData, setFormData] = useState({ item });

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    editTodo(id, formData.item);
    setEditing(false);
  };

  const editingTemplate = (
    <form onSubmit={handleSubmit}>
      <label htmlFor={`edit-${id}`}>Editing "{item}"</label>
      <input
        id={`edit-${id}`}
        name="item"
        type="text"
        value={formData.item}
        onChange={handleChange}
      />
      <button type="button" onClick={() => setEditing(false)}>
        Cancel
      </button>
      <button type="submit">Save</button>
    </form>
  );

  const viewTemplate = (
    <li>
      <input
        type="checkbox"
        id={id}
        checked={completed}
        onChange={() => toggleTaskCompleted(id)}
      />
      <label htmlFor={id}>{item}</label>

      {/* Edit button */}
      <button type="button" onClick={() => setEditing(true)}>
        Edit
      </button>

      {/* Remove button */}
      <button type="button" onClick={() => removeTodo(id)}>
        Remove
      </button>
    </li>
  );

  return isEditing ? editingTemplate : viewTemplate;
};

export default EditTodo;
