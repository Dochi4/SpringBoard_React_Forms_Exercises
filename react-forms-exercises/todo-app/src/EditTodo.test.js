import React from "react";
import { render, fireEvent } from "@testing-library/react";
import EditTodo from "./EditTodo";

it("renders without crashing", function () {
  render(<EditTodo />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<EditTodo />);
  expect(asFragment()).toMatchSnapshot();
});
