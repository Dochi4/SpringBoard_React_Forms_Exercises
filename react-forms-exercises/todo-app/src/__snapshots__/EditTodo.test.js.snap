// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li>
    <input
      type="checkbox"
    />
    <label />
    <button
      type="button"
    >
      Edit
    </button>
    <button
      type="button"
    >
      Remove
    </button>
  </li>
</DocumentFragment>
`;
